src/libraries/System.Net.Http/src/System/Net/Http/Headers/HttpContentHeaders.cs

```diff
@@ -110,6 +110,16 @@ internal HttpContentHeaders(HttpContent parent)
             : base(HttpHeaderType.Content | HttpHeaderType.Custom, HttpHeaderType.None)
         {
             _parent = parent;
+
+            if (parent != null)
+            {
+                // forces lazy loading so it can be consumed via foreach or via ToString()
+                try
+                {
+                    _ = ContentLength;
+                }
+                catch { }
```

Comment:
It's not acceptable to use try-catch for control flow and catch-all to swallow every exception.

Reply:
why is it not acceptable ? there a lot of examples https://github.com/search?q=repo%3Adotnet%2Fruntime+%22catch+%7B+%7D%22+language%3AC%23&type=code&l=C%23 of such pattern

I'm just trying to eagerly load a value (if I can) without introducing a breaking in HttpContentHeaders constructor

Reply:
If `TryComputeLength` is throwing we should propagate the exception. Something is going wrong in the user's code and we shouldn't hide that.

For example, this would make some of the problems mentioned in https://github.com/dotnet/runtime/pull/102416#issuecomment-2120245048 harder to notice.